src/tools/rbd_mirror/group_replayer/Replayer.cc

```diff
@@ -698,82 +804,131 @@ void Replayer<I>::create_group_snapshot(cls::rbd::GroupSnapshot snap,
   auto snap_type = cls::rbd::get_group_snap_namespace_type(
       snap.snapshot_namespace);
   if (snap_type == cls::rbd::GROUP_SNAPSHOT_NAMESPACE_TYPE_MIRROR) {
-    auto snap_ns = std::get_if<cls::rbd::GroupSnapshotNamespaceMirror>(
-        &snap.snapshot_namespace);
-    if (snap_ns->is_non_primary()) {
-      dout(10) << "remote group snapshot: " << snap.id << "is non primary"
+    auto snap_ns = std::get<cls::rbd::GroupSnapshotNamespaceMirror>(
```

Comment:
To avoid an unnecessary copy:
```suggestion
    const auto& snap_ns = std::get<cls::rbd::GroupSnapshotNamespaceMirror>(
```